work/worker.go

```diff
@@ -289,11 +295,27 @@ func (self *worker) update() {
 	quitByErr := make(chan bool, 1)
 	go self.handleTxsCh(quitByErr)
 
+	// Initialize restart watchdog
+	if self.nodetype == common.CONSENSUSNODE && self.restartFn != nil && self.restartTimeOut > 0 {
+		callback := func() {
```

Comment:
Do we need to wrap `restartFn` here? How about directly pass `self.restartFn` to `time.AfterFunc`?

Reply:
You are right. I updated the code like you want.